src/compiler/binder.ts

```diff
@@ -3701,6 +3706,8 @@ namespace ts {
                 break;
 
             case SyntaxKind.ComputedPropertyName:
+                // Computed property names are transformed by the ESNext transformer.
+                transformFlags |= TransformFlags.AssertESNext;
                 // Even though computed property names are ES6, we don't treat them as such.

```

Comment:
is this comment still accurate?

Reply:
Comment has been updated to reflect that ESNext is transforming some computed property names (even though they were first introduced in ES6)